pkg/rke2/serviceaccount_test.go

```diff
@@ -0,0 +1,131 @@
+package rke2
+
+import (
+	"context"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+var testServiceAccountEmpty = &v1.ServiceAccount{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "default",
+	},
+}
+var testServiceAccountFilled = &v1.ServiceAccount{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "default",
+	},
+	AutomountServiceAccountToken: new(bool),
```

Comment:
the `new` keyword is soon to be removed or deprecated from Go and the Go authors suggest not using it.

Reply:
vendor/k8s.io/utils/pointer `BoolPtr(bool)` perhaps?

Reply:
Dig it!

Reply:
Fixed!